src/commands/Misc/reddituser.ts

```diff
@@ -0,0 +1,219 @@
+			.catch(() => { throw message.language.tget('COMMAND_REDDITUSER_QUERY_FAILED'); }) as Reddit.Response<'comments'>;
+
+		for (const child of data.children) {
+			dataElements.push(child.data);
+		}
+
+		if (data.children.length === 100) await this.fetchComments(user, message, data.children[99].data.name, dataElements); // Recurse to get more data
+
+		return dataElements;
+	}
+
+	private async fetchSubmissions(user: string, message: KlasaMessage, after = '', dataElements: Reddit.PostDataElement[] = []) {
+		const url = new URL(`https://www.reddit.com/user/${user}/submitted.json`);
+		url.searchParams.append('after', after);
+		url.searchParams.append('limit', '100');
+
+		const { data } = await fetch(url, 'json')
+			.catch(() => { throw message.language.tget('COMMAND_REDDITUSER_QUERY_FAILED'); }) as Reddit.Response<'posts'>;
+
+		for (const child of data.children) {
+			dataElements.push(child.data);
+		}
+
+		if (data.children.length === 100) await this.fetchSubmissions(user, message, data.children[99].data.name, dataElements); // Recurse to get more data
+
+		return dataElements;
+	}
+
+	private calculateControversiality(comments: Reddit.CommentDataElement[]) {
+		if (!comments.length || comments.length < 5) return 0;
+		let count = 0;
+
+		for (const comment of comments) {
+			if (comment.controversiality === 1) count++;
+		}
+
+		return (count / comments.length) * 100;
+	}
+
+	private calculateKincaid(sentences: number, words: number, syllables: number) {
+		const sentenceWeight = 0.39;
+		const wordWeight = 11.8;
+		const adjustment = 15.59;
+
+		return (
+			(sentenceWeight * (words / sentences)) +
+			(wordWeight * (syllables / words)) -
+			adjustment
+		);
+	}
+
+	private calculateSyllables(word: string) {
+		word = word.toLowerCase();
+		if (word.length <= 3) return 1;
+		word = word.replace(/(?:[^laeiouy]es|ed|[^laeiouy]e)$/, '');
+		word = word.replace(/^y/, '');
+		const syl = word.match(/[aeiouy]{1,2}/g);
+
+		return syl ? syl.length : 1;
+	}
+
+	private calculateTextComplexity(comments: Reddit.CommentDataElement[]) {
+		let sentenceCount = 0;
+		let syllableCount = 0;
+		let wordCount = 0;
+
+		for (const comment of comments) {
+			const sentences = comment.body.split(/[.!?]+/gm);
+			const words = comment.body.trim().split(/\s+/gm);
+
+			sentenceCount += sentences.length - 1;
+			wordCount += words.length;
+
+			for (const syllable of words) {
+				syllableCount += this.calculateSyllables(syllable.toLowerCase());
+			}
+		}
+
+		return this.calculateKincaid(sentenceCount, wordCount, syllableCount);
+	}
+
+	private calculateTopSubredditsSubmissions(posts: Reddit.PostDataElement[]) {
+		const subredditCounts = [];
+		const subreddits: Record<string, number> = {};
+
+		for (const post of posts) {
+			subreddits[post.subreddit] = (subreddits[post.subreddit] || 0) + 1;
+		}
+
+		for (const [subreddit, count] of Object.entries(subreddits)) {
+			subredditCounts.push({
+				count,
+				name: subreddit
+			});
+		}
```

Comment:
This should use `Map<string, number>`, not `Record<string, number>`. Re-compiling and shape mutating are discouraged in this performance to avoid performance drawbacks.